SiriusXM with Ripping capability

kryptek opened the pull request:
Adds stream ripping and mp3 tagging capability. I've been using this to automatically record shows and push them into iCloud so they appear on my devices.

Commit:
Major refactor

Commit:
Fix authentication and layer issues

Search each layer in the marker lists for the episode layer because it's
not always guaranteed to be the same. Fixes the issue where the current
episode could not be found.

The authentication method now resets the session on login failures.
Previously the session would get borked and authentication would no
longer work on the session.

Commit:
Refactor episode polling

Past, current and future episodes are stored when retrieving the
now-playing data. As the current episode transitions, it is popped from
the store list of episodes until it's empty.

Commit:
Replace basic argv with argparse and remove static channel references

Commit:
Add ripping and refactoring

Commit:
Add delay when waiting for ffmpeg to stop and fix some logging errors

Commit:
Fix CRLF

Commit:
Add documentation and update argparse

Commit:
Add exception catch

Commit:
Add requirements

Commit:
Add my personal config as an example

Commit:
Update README.md

kryptek committed:
Update SXMAUTH cookie name

The cookie name appears to have changed to 'SXMAUTHNEW'. This change updates the cookie name so that the script understands it has authenticated.

mjgorman commented:
@kryptek I'm using your fork. Question about the episode recording. Should it stop/split between episodes? I'm using it to download from the Jason Ellis Channel, intent is to archive as many segments as i can for offline listening. It so far has not split, i'm just getting 1 large mp3 file. The guide info does seem correct in the app and online with different episode names.  Seems like something is off. i wonder if i need to tweak the data its looking at? 
```
18.May 2020 23:20:42 <SiriusXM>: Now Playing: Jason Ellis - The Future of Satellite Radio is the Jason Ellis Show. Every day, host Jason Ellis - a former pro skateboarder, pro fighter, and New York Times bestselling author - shares his humorous and irreverent worldview with the largest homegrown audience on SiriusXM. This is the uncensored talk show you’ve been waiting for your entire life. (22:39:17.434252 remaining)
```
There for sure isn't 22 hours left in this segmement/episode.

kryptek commented:
> @kryptek I'm using your fork. Question about the episode recording. Should it stop/split between episodes? I'm using it to download from the Jason Ellis Channel, intent is to archive as many segments as i can for offline listening. It so far has not split, i'm just getting 1 large mp3 file. The guide info does seem correct in the app and online with different episode names. Seems like something is off. i wonder if i need to tweak the data its looking at?
> 
> ```
> 18.May 2020 23:20:42 <SiriusXM>: Now Playing: Jason Ellis - The Future of Satellite Radio is the Jason Ellis Show. Every day, host Jason Ellis - a former pro skateboarder, pro fighter, and New York Times bestselling author - shares his humorous and irreverent worldview with the largest homegrown audience on SiriusXM. This is the uncensored talk show you’ve been waiting for your entire life. (22:39:17.434252 remaining)
> ```
> 
> There for sure isn't 22 hours left in this segmement/episode.

Hey, @mjgorman ,

It should be splitting between episodes, yes. It's odd that your episode shows as 22 hours remaining. Does the title from your XM Player ever change from the title you see in sxm.py ? The episode is pulled from the API's now-playing-live dict and that contains the episode start/end time. There must be another way we can detect episode changes for you, using longDescription/shortDescription or longTitle/mediumTitle.. 

I've also had thoughts about fetching the episode directly from the XM website. If we can't debug your issue, this may be something to look into.

mjgorman commented:
@kryptek Yeah was digging a bit more and seeing that the episode episodes are basically 24 hours long, looks like 12a-12a each day EST. The playing segment does change though. Jason Ellis's channel has segments that are anywhere from 20 to 50 minutes long usually so they are changing, here's an example from the app:

![Screenshot_20200519-162045](https://user-images.githubusercontent.com/5741377/82384348-f03bbd80-99ec-11ea-9c16-980a26ed0cca.png)
![Screenshot_20200519-162052](https://user-images.githubusercontent.com/5741377/82384352-f2058100-99ec-11ea-812a-697a36f765cd.png)

My goal is to split on those that way i can have a huge archive for when the wife and I are RVing without having to rely on cell data out boondocking. 

I did some lazy step debugging (added some extra logging statements) at line 256 in sxm.py right before it actually goes to process the episode data and I don't find those segment infos anywhere in the dict displayed.

mjgorman commented:
Heres the output from logging out the marker for the episode

```
mgorman@gormblade ~/sites/SiriusXM $ head output.log 
19.May 2020 10:06:39 <SiriusXM>: Recording the following shows
19.May 2020 10:06:39 <SiriusXM>: 	Jason Ellis
19.May 2020 10:06:39 <SiriusXM>: Automatic tagging data
19.May 2020 10:06:39 <SiriusXM>: 	Artist: The Jason Ellis Show | Album: Archived | Genre: Talk
19.May 2020 10:06:40 <SiriusXM>: EPISODE: {'legacyIds': {'shortId': '21687859'}, 'mediumTitle': 'Jason Ellis', 'longTitle': 'Jason Ellis', 'shortDescription': 'Funny, serious, unique & irreverent', 'longDescription': 'The Future of Satellite Radio is the Jason Ellis Show. Every day, host Jason Ellis - a former pro skateboarder, pro fighter, and New York Times bestselling author - shares his humorous and irreverent worldview with the largest homegrown audience on SiriusXM. This is the uncensored talk show you’ve been waiting for your entire life.', 'keywords': {}, 'episodeGUID': 'ecdfc8f6-e27a-0dad-616d-b9f8258deb9b', 'originalAirDate': '2020-05-19T04:00:00.000+0000', 'valuable': False, 'show': {'legacyIds': {'shortId': '6110'}, 'mediumTitle': 'Jason Ellis', 'longTitle': 'Jason Ellis', 'shortDescription': 'Funny, serious, unique & irreverent', 'longDescription': 'The Future of Satellite Radio is the Jason Ellis Show. Every day, host Jason Ellis - a former pro skateboarder, pro fighter, and New York Times bestselling author - shares his humorous and irreverent worldview with the largest homegrown audience on SiriusXM. This is the uncensored talk show you’ve been waiting for your entire life.', 'guid': 'db9613b2-ece6-407b-ba05-597286fef6cd', 'creativeArts': [{'encrypted': False, 'width': 1080, 'height': 1920, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/3b/330e2a-94e2-cab1-034f-b721cf9b7888.jpg', 'relativeUrl': '%Image%/images/chan/3b/330e2a-94e2-cab1-034f-b721cf9b7888.jpg', 'name': 'everest tile show background'}, {'encrypted': False, 'width': 1200, 'height': 1200, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/eb/b6e00c-e77e-a849-d9bd-de729fd3e222.png', 'relativeUrl': '%Image%/images/chan/eb/b6e00c-e77e-a849-d9bd-de729fd3e222.png', 'name': 'image'}, {'encrypted': False, 'width': 720, 'height': 720, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/1b/348a6e-8235-0101-6674-8a23bec62955.png', 'relativeUrl': '%Image%/images/chan/1b/348a6e-8235-0101-6674-8a23bec62955.png', 'name': 'image'}, {'encrypted': False, 'width': 360, 'height': 360, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/23/3c885f-c916-484f-4ef3-38a9056d7c82.png', 'relativeUrl': '%Image%/images/chan/23/3c885f-c916-484f-4ef3-38a9056d7c82.png', 'name': 'image'}, {'encrypted': False, 'width': 270, 'height': 270, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/97/2e1415-c04e-191c-5edf-cfde085d04fd.png', 'relativeUrl': '%Image%/images/chan/97/2e1415-c04e-191c-5edf-cfde085d04fd.png', 'name': 'image'}, {'encrypted': False, 'width': 240, 'height': 240, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/b5/1b2aaa-73d9-7085-0226-720efd409abe.png', 'relativeUrl': '%Image%/images/chan/b5/1b2aaa-73d9-7085-0226-720efd409abe.png', 'name': 'image'}, {'encrypted': False, 'width': 180, 'height': 180, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/50/529eb9-50e2-a62b-6930-3df9a0eb8e52.png', 'relativeUrl': '%Image%/images/chan/50/529eb9-50e2-a62b-6930-3df9a0eb8e52.png', 'name': 'image'}, {'encrypted': False, 'width': 120, 'height': 120, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/f5/5c4a06-8fdb-3bee-c3cb-081c923b6776.png', 'relativeUrl': '%Image%/images/chan/f5/5c4a06-8fdb-3bee-c3cb-081c923b6776.png', 'name': 'image'}, {'encrypted': False, 'width': 90, 'height': 90, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/c3/187eaf-6ef4-3ade-39ce-9c22a2aafd2a.png', 'relativeUrl': '%Image%/images/chan/c3/187eaf-6ef4-3ade-39ce-9c22a2aafd2a.png', 'name': 'image'}, {'encrypted': False, 'width': 60, 'height': 60, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/b5/8f1051-95f6-50f3-ac8f-4d75c7a65a28.png', 'relativeUrl': '%Image%/images/chan/b5/8f1051-95f6-50f3-ac8f-4d75c7a65a28.png', 'name': 'image'}, {'encrypted': False, 'width': 256, 'height': 320, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/5e/b07866-4c19-5b12-ff2b-77ef6fdee9dd.png', 'relativeUrl': '%Image%/images/chan/5e/b07866-4c19-5b12-ff2b-77ef6fdee9dd.png', 'name': 'show logo on dark'}, {'encrypted': False, 'width': 150, 'height': 150, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/0f/1fe9b6-fda6-cd52-e7c7-ffd238156f3c.png', 'relativeUrl': '%Image%/images/chan/0f/1fe9b6-fda6-cd52-e7c7-ffd238156f3c.png', 'name': 'show logo on dark'}, {'encrypted': False, 'width': 256, 'height': 320, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/ff/310f10-0e02-2f99-f195-9ce5d5cb9372.png', 'relativeUrl': '%Image%/images/chan/ff/310f10-0e02-2f99-f195-9ce5d5cb9372.png', 'name': 'show logo on light'}, {'encrypted': False, 'width': 150, 'height': 150, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/d4/7364d7-83ce-9131-4157-5144f85b9bb5.png', 'relativeUrl': '%Image%/images/chan/d4/7364d7-83ce-9131-4157-5144f85b9bb5.png', 'name': 'show logo on light'}, {'encrypted': False, 'width': 256, 'height': 320, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/46/7ec685-6fcf-2ebb-8f84-894d12869f3a.png', 'relativeUrl': '%Image%/images/chan/46/7ec685-6fcf-2ebb-8f84-894d12869f3a.png', 'name': 'show logo white'}, {'encrypted': False, 'width': 150, 'height': 150, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/2a/639593-c868-bd8b-f051-9d571b55437e.png', 'relativeUrl': '%Image%/images/chan/2a/639593-c868-bd8b-f051-9d571b55437e.png', 'name': 'show logo white'}, {'encrypted': False, 'width': 1920, 'height': 1920, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/f3/0e06c3-f649-9d8d-f8cd-729e06efceeb.jpg', 'relativeUrl': '%Image%/images/chan/f3/0e06c3-f649-9d8d-f8cd-729e06efceeb.jpg', 'name': 'square everest tile show background'}], 'showGUID': 'db9613b2-ece6-407b-ba05-597286fef6cd', 'connectInfo': {'facebook': ''}, 'disableRecommendations': ['false'], 'futureAirings': [{'timestamp': '2020-05-20T04:00:00.000+0000', 'channelId': '9363', 'duration': 86400}], 'aodEpisodeCount': 0, 'programType': 'Talk Show', 'isPlaceholderShow': True, 'isLiveVideoEligible': False, 'vodEpisodeCount': 0}, 'hot': False, 'highlighted': False, 'dmcaInfo': {'irNavClass': 'UNRESTRICTED_0', 'playOnSelect': 'REALTIME_6', 'channelContentType': 'T_3', 'maxFwdSkips': 5, 'maxBackSkips': 1, 'fwdSkipDur': 12, 'backSkipDur': 12, 'maxTotalSkips': 6, 'maxSkipDur': 3600000}, 'entities': {}, 'topics': {}, 'live': False, 'repeat': False}
19.May 2020 10:06:40 <SiriusXM>: EPISODE: {'episodeGUID': 'e878d99f-af63-0091-7042-444232e03f7a', 'show': {'guid': 'db9613b2-ece6-407b-ba05-597286fef6cd', 'showGUID': 'db9613b2-ece6-407b-ba05-597286fef6cd', 'aodEpisodeCount': 0, 'isLiveVideoEligible': False, 'vodEpisodeCount': 0}, 'dmcaInfo': {'irNavClass': 'UNRESTRICTED_0', 'playOnSelect': 'REALTIME_6', 'channelContentType': 'T_3', 'maxFwdSkips': 5, 'maxBackSkips': 1, 'fwdSkipDur': 12, 'backSkipDur': 12, 'maxTotalSkips': 6, 'maxSkipDur': 3600000}, 'live': False, 'repeat': False}
19.May 2020 10:06:40 <SiriusXM>: EPISODES: [{'mediumTitle': 'Jason Ellis', 'longTitle': 'Jason Ellis', 'shortDescription': 'Funny, serious, unique & irreverent', 'longDescription': 'The Future of Satellite Radio is the Jason Ellis Show. Every day, host Jason Ellis - a former pro skateboarder, pro fighter, and New York Times bestselling author - shares his humorous and irreverent worldview with the largest homegrown audience on SiriusXM. This is the uncensored talk show you’ve been waiting for your entire life.', 'start': datetime.datetime(2020, 5, 19, 4, 0), 'end': datetime.datetime(2020, 5, 20, 4, 0)}, {'mediumTitle': 'UnknownMediumTitle', 'longTitle': 'UnknownLongTitle', 'shortDescription': 'UnknownShortDescription', 'longDescription': 'UnknownLongDescription', 'start': datetime.datetime(2020, 5, 20, 4, 0), 'end': datetime.datetime(2020, 5, 21, 4, 0)}]
ffmpeg version 4.2.2 Copyright (c) 2000-2019 the FFmpeg developers

```

kryptek commented:
> Heres the output from logging out the marker for the episode
> 
> 
> 
> ```
> 
> mgorman@gormblade ~/sites/SiriusXM $ head output.log 
> 
> 19.May 2020 10:06:39 <SiriusXM>: Recording the following shows
> 
> 19.May 2020 10:06:39 <SiriusXM>: 	Jason Ellis
> 
> 19.May 2020 10:06:39 <SiriusXM>: Automatic tagging data
> 
> 19.May 2020 10:06:39 <SiriusXM>: 	Artist: The Jason Ellis Show | Album: Archived | Genre: Talk
> 
> 19.May 2020 10:06:40 <SiriusXM>: EPISODE: {'legacyIds': {'shortId': '21687859'}, 'mediumTitle': 'Jason Ellis', 'longTitle': 'Jason Ellis', 'shortDescription': 'Funny, serious, unique & irreverent', 'longDescription': 'The Future of Satellite Radio is the Jason Ellis Show. Every day, host Jason Ellis - a former pro skateboarder, pro fighter, and New York Times bestselling author - shares his humorous and irreverent worldview with the largest homegrown audience on SiriusXM. This is the uncensored talk show you’ve been waiting for your entire life.', 'keywords': {}, 'episodeGUID': 'ecdfc8f6-e27a-0dad-616d-b9f8258deb9b', 'originalAirDate': '2020-05-19T04:00:00.000+0000', 'valuable': False, 'show': {'legacyIds': {'shortId': '6110'}, 'mediumTitle': 'Jason Ellis', 'longTitle': 'Jason Ellis', 'shortDescription': 'Funny, serious, unique & irreverent', 'longDescription': 'The Future of Satellite Radio is the Jason Ellis Show. Every day, host Jason Ellis - a former pro skateboarder, pro fighter, and New York Times bestselling author - shares his humorous and irreverent worldview with the largest homegrown audience on SiriusXM. This is the uncensored talk show you’ve been waiting for your entire life.', 'guid': 'db9613b2-ece6-407b-ba05-597286fef6cd', 'creativeArts': [{'encrypted': False, 'width': 1080, 'height': 1920, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/3b/330e2a-94e2-cab1-034f-b721cf9b7888.jpg', 'relativeUrl': '%Image%/images/chan/3b/330e2a-94e2-cab1-034f-b721cf9b7888.jpg', 'name': 'everest tile show background'}, {'encrypted': False, 'width': 1200, 'height': 1200, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/eb/b6e00c-e77e-a849-d9bd-de729fd3e222.png', 'relativeUrl': '%Image%/images/chan/eb/b6e00c-e77e-a849-d9bd-de729fd3e222.png', 'name': 'image'}, {'encrypted': False, 'width': 720, 'height': 720, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/1b/348a6e-8235-0101-6674-8a23bec62955.png', 'relativeUrl': '%Image%/images/chan/1b/348a6e-8235-0101-6674-8a23bec62955.png', 'name': 'image'}, {'encrypted': False, 'width': 360, 'height': 360, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/23/3c885f-c916-484f-4ef3-38a9056d7c82.png', 'relativeUrl': '%Image%/images/chan/23/3c885f-c916-484f-4ef3-38a9056d7c82.png', 'name': 'image'}, {'encrypted': False, 'width': 270, 'height': 270, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/97/2e1415-c04e-191c-5edf-cfde085d04fd.png', 'relativeUrl': '%Image%/images/chan/97/2e1415-c04e-191c-5edf-cfde085d04fd.png', 'name': 'image'}, {'encrypted': False, 'width': 240, 'height': 240, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/b5/1b2aaa-73d9-7085-0226-720efd409abe.png', 'relativeUrl': '%Image%/images/chan/b5/1b2aaa-73d9-7085-0226-720efd409abe.png', 'name': 'image'}, {'encrypted': False, 'width': 180, 'height': 180, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/50/529eb9-50e2-a62b-6930-3df9a0eb8e52.png', 'relativeUrl': '%Image%/images/chan/50/529eb9-50e2-a62b-6930-3df9a0eb8e52.png', 'name': 'image'}, {'encrypted': False, 'width': 120, 'height': 120, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/f5/5c4a06-8fdb-3bee-c3cb-081c923b6776.png', 'relativeUrl': '%Image%/images/chan/f5/5c4a06-8fdb-3bee-c3cb-081c923b6776.png', 'name': 'image'}, {'encrypted': False, 'width': 90, 'height': 90, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/c3/187eaf-6ef4-3ade-39ce-9c22a2aafd2a.png', 'relativeUrl': '%Image%/images/chan/c3/187eaf-6ef4-3ade-39ce-9c22a2aafd2a.png', 'name': 'image'}, {'encrypted': False, 'width': 60, 'height': 60, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/b5/8f1051-95f6-50f3-ac8f-4d75c7a65a28.png', 'relativeUrl': '%Image%/images/chan/b5/8f1051-95f6-50f3-ac8f-4d75c7a65a28.png', 'name': 'image'}, {'encrypted': False, 'width': 256, 'height': 320, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/5e/b07866-4c19-5b12-ff2b-77ef6fdee9dd.png', 'relativeUrl': '%Image%/images/chan/5e/b07866-4c19-5b12-ff2b-77ef6fdee9dd.png', 'name': 'show logo on dark'}, {'encrypted': False, 'width': 150, 'height': 150, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/0f/1fe9b6-fda6-cd52-e7c7-ffd238156f3c.png', 'relativeUrl': '%Image%/images/chan/0f/1fe9b6-fda6-cd52-e7c7-ffd238156f3c.png', 'name': 'show logo on dark'}, {'encrypted': False, 'width': 256, 'height': 320, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/ff/310f10-0e02-2f99-f195-9ce5d5cb9372.png', 'relativeUrl': '%Image%/images/chan/ff/310f10-0e02-2f99-f195-9ce5d5cb9372.png', 'name': 'show logo on light'}, {'encrypted': False, 'width': 150, 'height': 150, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/d4/7364d7-83ce-9131-4157-5144f85b9bb5.png', 'relativeUrl': '%Image%/images/chan/d4/7364d7-83ce-9131-4157-5144f85b9bb5.png', 'name': 'show logo on light'}, {'encrypted': False, 'width': 256, 'height': 320, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/46/7ec685-6fcf-2ebb-8f84-894d12869f3a.png', 'relativeUrl': '%Image%/images/chan/46/7ec685-6fcf-2ebb-8f84-894d12869f3a.png', 'name': 'show logo white'}, {'encrypted': False, 'width': 150, 'height': 150, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/2a/639593-c868-bd8b-f051-9d571b55437e.png', 'relativeUrl': '%Image%/images/chan/2a/639593-c868-bd8b-f051-9d571b55437e.png', 'name': 'show logo white'}, {'encrypted': False, 'width': 1920, 'height': 1920, 'type': 'IMAGE', 'url': 'http://pri.art.prod.streaming.siriusxm.com/images/chan/f3/0e06c3-f649-9d8d-f8cd-729e06efceeb.jpg', 'relativeUrl': '%Image%/images/chan/f3/0e06c3-f649-9d8d-f8cd-729e06efceeb.jpg', 'name': 'square everest tile show background'}], 'showGUID': 'db9613b2-ece6-407b-ba05-597286fef6cd', 'connectInfo': {'facebook': ''}, 'disableRecommendations': ['false'], 'futureAirings': [{'timestamp': '2020-05-20T04:00:00.000+0000', 'channelId': '9363', 'duration': 86400}], 'aodEpisodeCount': 0, 'programType': 'Talk Show', 'isPlaceholderShow': True, 'isLiveVideoEligible': False, 'vodEpisodeCount': 0}, 'hot': False, 'highlighted': False, 'dmcaInfo': {'irNavClass': 'UNRESTRICTED_0', 'playOnSelect': 'REALTIME_6', 'channelContentType': 'T_3', 'maxFwdSkips': 5, 'maxBackSkips': 1, 'fwdSkipDur': 12, 'backSkipDur': 12, 'maxTotalSkips': 6, 'maxSkipDur': 3600000}, 'entities': {}, 'topics': {}, 'live': False, 'repeat': False}
> 
> 19.May 2020 10:06:40 <SiriusXM>: EPISODE: {'episodeGUID': 'e878d99f-af63-0091-7042-444232e03f7a', 'show': {'guid': 'db9613b2-ece6-407b-ba05-597286fef6cd', 'showGUID': 'db9613b2-ece6-407b-ba05-597286fef6cd', 'aodEpisodeCount': 0, 'isLiveVideoEligible': False, 'vodEpisodeCount': 0}, 'dmcaInfo': {'irNavClass': 'UNRESTRICTED_0', 'playOnSelect': 'REALTIME_6', 'channelContentType': 'T_3', 'maxFwdSkips': 5, 'maxBackSkips': 1, 'fwdSkipDur': 12, 'backSkipDur': 12, 'maxTotalSkips': 6, 'maxSkipDur': 3600000}, 'live': False, 'repeat': False}
> 
> 19.May 2020 10:06:40 <SiriusXM>: EPISODES: [{'mediumTitle': 'Jason Ellis', 'longTitle': 'Jason Ellis', 'shortDescription': 'Funny, serious, unique & irreverent', 'longDescription': 'The Future of Satellite Radio is the Jason Ellis Show. Every day, host Jason Ellis - a former pro skateboarder, pro fighter, and New York Times bestselling author - shares his humorous and irreverent worldview with the largest homegrown audience on SiriusXM. This is the uncensored talk show you’ve been waiting for your entire life.', 'start': datetime.datetime(2020, 5, 19, 4, 0), 'end': datetime.datetime(2020, 5, 20, 4, 0)}, {'mediumTitle': 'UnknownMediumTitle', 'longTitle': 'UnknownLongTitle', 'shortDescription': 'UnknownShortDescription', 'longDescription': 'UnknownLongDescription', 'start': datetime.datetime(2020, 5, 20, 4, 0), 'end': datetime.datetime(2020, 5, 21, 4, 0)}]
> 
> ffmpeg version 4.2.2 Copyright (c) 2000-2019 the FFmpeg developers
> 
> 
> 
> ```

I see a couple of different guid’s in your log output.. any chance those are changing when the segments change? fef6cd seems like it might be specific to the “episode”; I’d guess that maybe e03f7a or 8deb9b might change somewhere when the segments change. Any chance you can check that out?

kryptek committed:
Fix broken channel list

Sirius recently updated their API and changed the JSON structure around a bit.
This change fixes the functions which access the channel list response so that
the script continues to work

kryptek committed:
Remove time parameter from channel list uri

kryptek committed:
Make config files optional
